### ISSUE TITLE
Example Usage on ReadMe.md

### ISSUE THREAD

**Issue by ohenepee** [tusharmath/Multi-threaded-downloader#194]
I thought adding an example usage _(e.g. downloading the Node Runtime from nodejs.org)_ was essential to every library... b'coz I can't find any example of how to use this seemingly awesome library.

**Comment by minhnhut** [comment]
Agree, I'm still trying to figure out how to use this library properly in my system. I end up with using zeltice-mt-downloader- a fork of this library.

**Comment by ProtocolNebula** [comment]
I need a nice example, I'm using this code: https://github.com/tusharmath/Multi-threaded-downloader/issues/166#issuecomment-282952823 (modified to add my progress bar and other things), but its too much ugly and hard to read/use.

It was hard to find and use, a good example will be great.

**Comment by minhnhut** [comment]
> I need a nice example, I'm using this code: [#166 (comment)](https://github.com/tusharmath/Multi-threaded-downloader/issues/166#issuecomment-282952823) (modified to add my progress bar and other things), but its too much ugly and hard to read/use.
> 
> It was hard to find and use, a good example will be great.

After investigating. I still feel impossible to easily integrate the library to my project. So I end up wrote my own downloader solution. I don't mean to advertisement. If you don't mind. You can try my [rapid-downloader](https://github.com/minhnhut/rapid-downloader) package, it is available on NPM.

**Comment by ProtocolNebula** [comment]
> > I need a nice example, I'm using this code: [#166 (comment)](https://github.com/tusharmath/Multi-threaded-downloader/issues/166#issuecomment-282952823) (modified to add my progress bar and other things), but its too much ugly and hard to read/use.
> > It was hard to find and use, a good example will be great.
> 
> After investigating. I still feel impossible to easily integrate the library to my project. So I end up wrote my own downloader solution. I don't mean to advertisement. If you don't mind. You can try my [rapid-downloader](https://github.com/minhnhut/rapid-downloader) package, it is available on NPM.

Thanks, I was just about to make my own downloader or at least a wrapper (in fact, I have a wrapper with this project to add missing elements and make it easy usable).

I will take a look to your project, I hope it do all I need or almost.

**Comment by muscaiu** [comment]
It's the first node library that i don't know how to use lol.
Weird documentation dude.
Write examples.
